README.md

```diff
@@ -145,7 +187,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+refer to [readme.txt](readme.txt) to run properly
```

Comment:
> "the big cat was owned by joe exotic"

LOL

OK, so I'm genuinely impressed. Your script worked beautifully, and all tests passed here. Your code is table driven, you have positive and negative tests: everything is fine.

# +5

You're right: the code is a bit too imperative to be good Prolog style, but it's still pretty cool.

Here's my attempt:

~~~ prolog
valid_sentence(S) :- fsm(0, S).

fsm(0, [ the | Rest ])     :- fsm(1, Rest).
fsm(1, [ lazy | Rest ])    :- fsm(1, Rest).
fsm(1, [ smelly | Rest ])  :- fsm(1, Rest).
fsm(1, [ cat | Rest ])     :- fsm(2, Rest).
fsm(1, [ dog | Rest ])     :- fsm(2, Rest).
fsm(2, [ ate | Rest ])     :- fsm(3, Rest).
fsm(2, [ ran | Rest ])     :- fsm(3, Rest).
fsm(3, [ eoi ])            :- !.
fsm(3, [ slowly | Rest ])  :- fsm(4, Rest).
fsm(3, [ noisily | Rest ]) :- fsm(4, Rest).
fsm(4, [ eoi ])            :- !.

state_machine(S) :- valid_sentence(S).

...

test(state_machine) :- state_machine([the, cat, ate, eoi]).
~~~